### ISSUE TITLE
Proxy

### ISSUE THREAD

**Issue by cblp** [vadimvinnik/empty-monad#1]
https://hackage.haskell.org/package/base/docs/Data-Proxy.html

**Comment by vadimvinnik** [comment]
Yes, you are right. When I came to this idea, I tried to find something similar on Hackage - because the idea is too obvious. I was searching for terms like empty, null, nothing, void etc and did not find anything. Much later, after having uploaded my implementation, I learned about Proxy - this metaphor does not seem obvious.

**Comment by cblp** [comment]
Proxy is often (mainly?) used as a proxy between types and values worlds. Hence the name.
